### PR TITLE
BS-13494 Prevent use case non empty Process Contract combined with something else thant "none" start event

### PR DESCRIPTION
**Pull request by rbioteau**
Prevent use case non empty Process Contract combined with something else
thant "none" start event

### CODE REVIEW

File: bundles/plugins/org.bonitasoft.studio.tests-utils/src/org/bonitasoft/studio/model/process/builders/StartMessageEventBuilder.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright (C) 2014 BonitaSoft S.A.
```
Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.tests-utils/src/org/bonitasoft/studio/model/process/builders/StartMessageEventBuilder.java]: 2015 BonitaSoft

File: bundles/plugins/org.bonitasoft.studio.tests-utils/src/org/bonitasoft/studio/model/process/builders/StartSignalEventBuilder.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright (C) 2014 BonitaSoft S.A.
```
Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.tests-utils/src/org/bonitasoft/studio/model/process/builders/StartSignalEventBuilder.java]: 2015 BonitaSoft

File: bundles/plugins/org.bonitasoft.studio.validation/messages.properties [hunk 3]
```diff
@@ -104,4 +104,5 @@ invalidMultipleFileContractInput=Invalid initial content for document {0}. A sin
 invalidFileContractInputType=Invalid initial content for document {0}. A FILE contract input is expected but a {1} contract input has been found.
 invalidSingleFileContractInput=Invalid initial content for document {0}. A multiple FILE contract input is expected but a single FILE contract input has been found.
 missingFileContractInput=Invalid initial content for document {0}. No FILE contract input has been found.
-emptyFormMappingWarning=No target form is defined. An autogenerated form will be used in the development environment ONLY.
\ No newline at end of file
+emptyFormMappingWarning=No target form is defined. An autogenerated form will be used in the development environment ONLY.
+cannotUseThisStartEventTypeWithAContract=It is not supported to use an instantiation contract with this start event type (only "none" start event is supported).
```
Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.validation/messages.properties]: did you reviewed the text with Chris?

File: bundles/plugins/org.bonitasoft.studio.validation/src/org/bonitasoft/studio/validation/constraints/process/StartEventWithContractConstraint.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+import org.bonitasoft.studio.model.process.StartTimerEvent;
+import org.bonitasoft.studio.model.process.util.ProcessSwitch;
+import org.bonitasoft.studio.validation.constraints.AbstractLiveValidationMarkerConstraint;
+import org.bonitasoft.studio.validation.i18n.Messages;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.validation.IValidationContext;
+
+/**
+ * @author Baptiste Mesta
```
Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.validation/src/org/bonitasoft/studio/validation/constraints/process/StartEventWithContractConstraint.java]: author Baptiste?

File: bundles/plugins/org.bonitasoft.studio.validation/src/org/bonitasoft/studio/validation/constraints/process/StartEventWithContractConstraint.java [hunk 5]
```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright (C) 2009 BonitaSoft S.A.
```
Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.validation/src/org/bonitasoft/studio/validation/constraints/process/StartEventWithContractConstraint.java]: 2009 BonitaSoft

File: bundles/plugins/org.bonitasoft.studio.validation/src-test/java/org/bonitasoft/studio/validation/constraints/process/StartEventWithContractConstraintTest.java [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+        final StartEventWithContractConstraint constraint = new StartEventWithContractConstraint();
+
+        final IValidationContext ctx = aValidationContext(aStartTimerEvent().in(aPool().havingContract(aContract())).build());
+        constraint.performBatchValidation(ctx);
+
+        verify(ctx).createSuccessStatus();
+    }
+
+    @Test
+    public void should_not_fail_if_process_have_a_none_timer_event_and_has_a_contract() throws Exception {
```
Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.validation/src-test/java/org/bonitasoft/studio/validation/constraints/process/StartEventWithContractConstraintTest.java]: have a none event instad of a none timer event?

File: bundles/plugins/org.bonitasoft.studio.validation/messages.properties (hunk 3 above)
Comment (rbioteau) [inline bundles/plugins/org.bonitasoft.studio.validation/messages.properties]: No

**Comment by apupier** [comment]
OK for merge after text reviewed by an english speaker

**Comment by rbioteau** [comment]
Message reviewed

**Pull request merged**